### PR TITLE
Add tests and solve N+1 problems

### PR DESCRIPTION
**Pull request by tarek102**
## In this PR:

- Solve N+1 problems.
- Create spec tests for User index page.
- Create spec tests for User show page.
- Create spec tests for Post index page.
- Create spec tests for Post showpage.

### CODE REVIEW

File: spec/view/post_index_spec.rb [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    )
+
+    Comment.create(post_id: @post1.id, author_id: @user.id, text: 'Great job!')
+    Comment.create(post_id: @post2.id, author_id: @user.id, text: 'Thanks for the effort')
+    Like.create(post_id: @post1.id, author_id: @user.id)
+    Like.create(post_id: @post2.id, author_id: @user.id)
+
+    visit user_posts_path(@user)
+  end
+
```
Comment (VICTOR-CHUKWU) [inline spec/view/post_index_spec.rb]: - [ ] kindly fix your test as some are failing
 
image

**Review by VICTOR-CHUKWU — CHANGES_REQUESTED** [review]
Hi @tarek102  👋  

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

almost

## Highlights
- Good PR
- Good commit
- All linters passed


## Required Changes ♻️

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/view/post_index_spec.rb (hunk 1 above)
Comment (tarek102) [reply spec/view/post_index_spec.rb]: @VICTOR-CHUKWU Thank you for your reply, I am having the tests pass on my pc 
image

Can you send me a screenshot of one of the failing tests, to know the error?

Comment (VICTOR-CHUKWU) [reply spec/view/post_index_spec.rb]: @tarek102  I hope its clear now after the zoom call, Kindly make the changes and the next reviewer should approve :clap:

Comment (tarek102) [reply spec/view/post_index_spec.rb]: Thanks for the zoom meeting, really appreciate it.

**Review by Davidosky007 — APPROVED** [review]
Hi @tarek102 @gak92 👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
* No linter errors 👍 
* All tests passed ✔️ 
* Proper use of Gitflow 👍 
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**